plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/transport/refresh/TransportRefreshAction.java

```diff
@@ -0,0 +1,117 @@
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.TransportAction;
+import org.elasticsearch.cluster.ClusterService;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.plugin.ingest.PipelineStore;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Refreshes the in-memory representaion of all the pipelines on all ingest nodes.
+ */
+public class TransportRefreshAction extends TransportAction<RefreshRequest, RefreshResponse> implements TransportRequestHandler<RefreshRequest> {
+
+    private final PipelineStore pipelineStore;
+    private final ClusterService clusterService;
+    private final TransportService transportService;
+
+    @Inject
+    public TransportRefreshAction(Settings settings, ThreadPool threadPool, ActionFilters actionFilters, IndexNameExpressionResolver resolver, PipelineStore pipelineStore, ClusterService clusterService, TransportService transportService) {
+        super(settings, RefreshAction.NAME, threadPool, actionFilters, resolver);
+        this.pipelineStore = pipelineStore;
+        this.clusterService = clusterService;
+        this.transportService = transportService;
+        transportService.registerRequestHandler(RefreshAction.NAME, RefreshRequest::new, ThreadPool.Names.SAME, this);
+    }
+
+
+    @Override
+    protected void doExecute(RefreshRequest request, ActionListener<RefreshResponse> listener) {
+        List<DiscoveryNode> ingestNodes = new ArrayList<>();
+        for (DiscoveryNode node : clusterService.state().getNodes()) {
+            String nodeEnabled = node.getAttributes().get("ingest");
+            if ("true".equals(nodeEnabled)) {
+                ingestNodes.add(node);
+            }
+        }
+
+        if (ingestNodes.isEmpty()) {
+            throw new IllegalStateException("There are no ingest nodes in this cluster");
+        }
+
+        AtomicBoolean failed = new AtomicBoolean();
+        AtomicInteger expectedResponses = new AtomicInteger(ingestNodes.size());
+        for (DiscoveryNode node : ingestNodes) {
+            RefreshRequest nodeRequest = new RefreshRequest();
+            transportService.sendRequest(node, RefreshAction.NAME, nodeRequest, new TransportResponseHandler<RefreshResponse>() {
+                @Override
+                public RefreshResponse newInstance() {
+                    return new RefreshResponse();
+                }
+
+                @Override
+                public void handleResponse(RefreshResponse response) {
+                    decrementAndReturn();
```

Comment:
we ignore whether refresh failed or not. (cause it's at the moment always true) Wondering if we should take it into account.